gamms/VisualizationEngine/graph_visual.py

```diff
@@ -14,6 +14,90 @@ def __init__(self, graph, width=1980, height=1080):
         self.y_max = max(node.y for node in self.graph.nodes.values())
         self.offset = (0.0, 0.0)
 
+        # Data Storage. INTERNAL USE ONLY
+        self.__edge_list = []
+        self.__node_list = []
+        self.__node_color_list = {}
+        self.__edge_color_list = {}
+        self.__node_size_list = {}
+        self.__edge_size_list = {}
+
+        # Set the data stores. 
+        self._initalize_data()
+
+    # Internal Methouds
+    def _initalize_data(self) -> None:
+        for edge in self.graph.edges.values():
+            self.__edge_list.append(edge.id)
+
+        for node in self.graph.nodes.values():
+            self.__node_list.append(node.id)
+
+    # Color Set Methouds
+    def setNodeColor(self, node, color = (255, 255, 255)):
+        if node.id in self.__node_list:
+            self.__node_color_list[node.id] = color
+        else:
+            print(f"Warning: setNodeColorByID({node.id}) not found.")
+
+    def setEdgeColor(self, source_id, target_id, color = (255, 255, 255)):
+        """Edge id is a bitwise encoding."""
+        edge_id = (source_id << 32) + target_id
+        if edge_id is self.__edge_list:
```

Comment:
I believe that should be `if edge_id in self.__edge_list:`?
```suggestion
        if edge_id in self.__edge_list:
```

Reply:
Yes, it should be.